### ISSUE TITLE
Incorrect signature error when verifying the signature of an utf8-encoded SAML response

### ISSUE THREAD

**Issue by kiejo** [tngan/samlify#150]
I get an "incorrect signature" error when verifying a response with a signed assertion that contains utf8-encoded characters in an `AttributeValue`.
I was able to track down the issue to the `base64Decode` function, which uses `ASCII` to encode the base64 decoded data:
https://github.com/tngan/samlify/blob/c71f9f6729d256383a689d5aa79e8c1af08bd7bc/src/utility.ts#L30
Changing it to use the default encoding (UTF-8) fixes the issue:
```typescript
return Boolean(isBytes) ? bytes : bytes.toString();
```

What is the reason that the function currently uses `ASCII` and not `UTF-8` encoding? I discovered this issue when testing this library against responses from Google Suite SAML, which can contain such characters. If there is no good reason to use `ASCII` it would be great to switch it to `UTF-8` to fix this issue.

---
I created the following example response using https://www.samltool.com/sign_response.php and choosing "Sign Assertion". Note that the `first_name` field contains the text `Björn`, which is what causes the error.
```xml
<samlp:Response xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" ID="_8e8dc5f69a98cc4c1ff3427e5ce34606fd672f91e6" Version="2.0" IssueInstant="2017-11-02T16:17:29Z" Destination="https://api.nuclino.dev:4000/api/sso/bbc5f56e-4a50-4707-b5ae-161c26bbf5f0/acs">
  <saml:Issuer>https://test.issuer.com/sso</saml:Issuer>
  <samlp:Status>
    <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
  </samlp:Status>
  <saml:Assertion xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xs="http://www.w3.org/2001/XMLSchema" ID="pfx94b457ae-3471-8904-bcc3-8d8cf62bb34e" Version="2.0" IssueInstant="2017-11-02T16:17:29Z">
    <saml:Issuer>https://test.issuer.com/sso</saml:Issuer><ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
  <ds:SignedInfo><ds:CanonicalizationMethod Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/>
    <ds:SignatureMethod Algorithm="http://www.w3.org/2000/09/xmldsig#rsa-sha1"/>
  <ds:Reference URI="#pfx94b457ae-3471-8904-bcc3-8d8cf62bb34e"><ds:Transforms><ds:Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/><ds:Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/></ds:Transforms><ds:DigestMethod Algorithm="http://www.w3.org/2000/09/xmldsig#sha1"/><ds:DigestValue>5gA7i6yuNEOVYO1AB2BEBG0YNqg=</ds:DigestValue></ds:Reference></ds:SignedInfo><ds:SignatureValue>QI1SP9NQmvJBPpzHpjQ5mX+e6396wt6mTjRMl6qWd/eB/r8iSJJjiQC+BQ5M24k1JPmacBzrL5qkBa8NQPn3pWdKGxbr35lGxqY3CvDHbzZb8T6oMbr0uivleTlkBbJOZqKdcYUpASwu+NS2RoXUDsxMxYbe6e5UG7NpyZ0L8v8=</ds:SignatureValue>
<ds:KeyInfo><ds:X509Data><ds:X509Certificate>MIICVDCCAb2gAwIBAgIBADANBgkqhkiG9w0BAQ0FADBHMQswCQYDVQQGEwJkZTEQMA4GA1UECAwHQmF2YXJpYTEQMA4GA1UECgwHTnVjbGlubzEUMBIGA1UEAwwLbnVjbGluby5jb20wHhcNMTcxMTAzMTUxMjMwWhcNMjcxMTAxMTUxMjMwWjBHMQswCQYDVQQGEwJkZTEQMA4GA1UECAwHQmF2YXJpYTEQMA4GA1UECgwHTnVjbGlubzEUMBIGA1UEAwwLbnVjbGluby5jb20wgZ8wDQYJKoZIhvcNAQEBBQADgY0AMIGJAoGBAK05fGI0ZnakEFLxdIrX2ECVdAa+OXtuh1/qedoY8vg+GJMv7NLb9hzKTr2okUF0YB1rUpiQOed8HqtAlhNmtiMHG/zKG2CBuDj544iRJ0/+DpFsaWUCslTn30xf3kcjmL7YRIcJeEbmGlrjRVSLJNS1tQTNxqi6dQfdbSQ6TyBvAgMBAAGjUDBOMB0GA1UdDgQWBBR7t8ngeOUmnNuplBDFqiGDFWoTITAfBgNVHSMEGDAWgBR7t8ngeOUmnNuplBDFqiGDFWoTITAMBgNVHRMEBTADAQH/MA0GCSqGSIb3DQEBDQUAA4GBAJgvEvdEwsLNw5WZfDqa7C0Zghc+e05p+PEQOL80sq8PRSFY0A5B6toy+ijnCZJbVTPmOM4w9fA25sihrW2bh84HOMrwFyy+HmUjO1nBoU/P99fZcJGbcosYr3cJCyLxVR8rF4X4BkKnStmoxOVu2lbhQGN9Gr9ew61T1bmspXIZ</ds:X509Certificate></ds:X509Data></ds:KeyInfo></ds:Signature>
    <saml:Subject>
      <saml:NameID Format="urn:oasis:names:tc:SAML:2.0:nameid-format:transient">bjorn.doe@nuclino.com</saml:NameID>
    </saml:Subject>
    <saml:Conditions NotBefore="2017-11-02T16:17:29Z" NotOnOrAfter="2024-01-18T06:21:48Z">
      <saml:AudienceRestriction>
        <saml:Audience>https://api.nuclino.dev:4000/api/sso/bbc5f56e-4a50-4707-b5ae-161c26bbf5f0/metadata</saml:Audience>
      </saml:AudienceRestriction>
    </saml:Conditions>
    <saml:AttributeStatement>
      <saml:Attribute Name="uid" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
        <saml:AttributeValue xsi:type="xs:string">9e11b1b8-8d8b-4119-9786-41e04194e197</saml:AttributeValue>
      </saml:Attribute>
      <saml:Attribute Name="first_name" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
        <saml:AttributeValue xsi:type="xs:string">Björn</saml:AttributeValue>
      </saml:Attribute>
      <saml:Attribute Name="last_name" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
        <saml:AttributeValue xsi:type="xs:string">Doe</saml:AttributeValue>
      </saml:Attribute>
    </saml:AttributeStatement>
  </saml:Assertion>
</samlp:Response>
```

**Comment by tngan** [comment]
@kiejo +1 for your nice catch and detailed report. Yes, utf-8 is definitely the better choice.

**Comment by kiejo** [comment]
Nice, thanks for the fast response and merge!

**Comment by tngan** [comment]
@kiejo  Patched with v2.3.6 and I have closed this issue. Thanks for your contribution.
